Tools/autotest/arduplane.py

```diff
@@ -1164,6 +1164,55 @@ def ThrottleFailsafeFence(self):
             raise NotAchievedException("Fence not enabled after RC fail")
         self.do_fence_disable() # Ensure the fence is disabled after test
 
+    def GCSFailsafe(self):
+        '''Ensure Long-Failsafe works on GCS loss'''
+        self.start_subtest("Test Failsafe: RTL")
+        self.load_sample_mission()
+        self.set_parameter("RTL_AUTOLAND", 1)
+        self.change_mode("AUTO")
+        self.takeoff()
+        self.set_parameter("FS_GCS_ENABL", 1)
+        self.set_parameter("FS_LONG_ACTN", 1)
+        self.progress("Disconnecting GCS")
+        self.set_heartbeat_rate(0)
+        self.delay_sim_time(5)
```

Comment:
Shoudln't need to `delay_sim_time` before `wait_mode`.  If you need to wait for longer there's a timeout parameter to `wait_mode`

Reply:
Ah, yes. That makes sense, thanks for pointing that out.